src/Nuxeo/Client/Api/NuxeoClient.php

```diff
@@ -68,16 +68,39 @@ class NuxeoClient {
    * @param string $password
    * @throws NuxeoClientException
    */
-  public function __construct($url = 'http://localhost:8080/nuxeo', $username = 'Administrator', $password = 'Administrator') {
+  public function __construct($url = 'http://localhost:8080/nuxeo', $username = 'Administrator', $password = 'Administrator', $authType = 'BASIC', $portalKey = "") {
```

Comment:
Using a dedicated interceptor for authentication would avoid too much complexity in the constructor.